extras/module-node/duk_module_node.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Node.js-like require() and module loader
+ *
+ * https://nodejs.org/api/modules.html
+ */
+
+#include "duktape.h"
+#include "duk_module_node.h"
+
+static duk_int_t duk__eval_module_helper (duk_context *ctx);
+static void duk__push_module_object (duk_context *ctx, const char *id);
+
+static duk_resolve_module_function resolve_module_cb;
+static duk_load_module_function load_module_cb;
+
+static duk_bool_t duk__get_cached_module(duk_context *ctx, const char *id) {
+	duk_push_global_stash(ctx);
+	duk_get_prop_string(ctx, -1, "\xffModuleCache");
```

Comment:
The internal key here (and in other places) follows the format reserved for Duktape internals. Extras are maybe a bit of a grey area in this respect, so I'm not sure what'd be best.